### DISCUSSION TITLE
Scrapers question.

### DISCUSSION THREAD

**Discussion by DROIDshield** [jurialmunkey/skin.arctic.horizon.2#920] [Discussion]
Is there any other scrapers I could add to kodi to boost scraping as within the  skin widgets, not all meta data are being scraped and I might be missing posters, clear art etc.

Any ideas?

**Comment by jurialmunkey** [comment]
Scrapers only matter for content in your library and widgets based on library content.

**Reply by DROIDshield** [reply]
So forgot content TMDB is looking at and some af the meta is not there fully is down to the server it is on and what detail that library holds? 

Nothing can be done about that im guessing?

**Reply by jurialmunkey** [reply]
> content TMDb is looking at

What does this mean? 

TMDb is a website. It does not look at your Kodi information.

Are you talking about the TMDb library scraper? TMDbHelper plugin? TMDbHelper service? 
Unclear what you're referring to here.

> meta is not there fully is down to the server it is on

What do you mean by "the server it is on"?

Widgets from plugins use the metadata the plugin supplies
Widgets from library use the metadata the library supplies

Metadata is just plain text attached to the listitem by the library or the plugin supplying it.

**How a library scraper or plugin works**

- When you scan the library, the library scraper looks at all your source folders and reads the filenames. 
- It then "scrapes" information from various websites and APIs for info it thinks matches the item the filename describes. 
- That info is formatted into plain text metadata which is saved to the library database. 

Plugins function in a similar way except that they store metadata in their own personal database/cache and might do the "scraping" for the items that they want to display on a more "on-demand" basis (i.e. when the plugin is "loading" before the items appear).

**How a widget appears on screen**

- The skin defines a library/plugin path which Kodi calls
- The library/plugin simply attaches the appropriate metadata to the listitems and then passes them to Kodi as a list.
- The library/plugin then exits before the items appear on screen

There is no "server" that Kodi is dynamically retrieving metadata from - the information displayed is simply plain text that the plugin supplied before it exited.

**How an image appears on screen**

- The skin defines an "itemlayout" which is a series of texture and label controls with coordinates
- The texture control might have a texture defined in an infolabel e.g. $INFO[ListItem.Art(poster)]
- That infolabel will contain plain text metadata from the library/plugin - likely an image URL to a local file or website.
- Kodi downloads the image, saves it to cache, and then displays it on screen according to the coordinates the skin defines in the texture control.

**Reply by DROIDshield** [reply]
Honestly I had no idea this is how it works. I thought when I click onto a widget in TMDB it was pulling meta from a server.

I'm a novis at this and just finding my way though the woods.

The info you have laid out above is just what I needed for this to be understood and sink into my head.

Thank you so much for the info passed on.

You should have a YouTube channel doing tutorials. I swear you would get the hits and the revenue. Just saying.🤗👍
